### PR TITLE
ensure creation of .dirmeta files

### PR DESCRIPTION
**Pull request by ryanfkeepers**
the tree wasn't properly populating the dirmeta file for folders.  This change is primarily for the addition of that file, but also contains a minor refactor for taking in the folder drive item instead of only getting its properties.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :bug: Bugfix

#### Issue(s)

* #4689

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/m365/collection/drive/delta_tree.go [hunk 1]
```diff
@@ -66,7 +67,7 @@ func newFolderyMcFolderFace(
 
 // reset erases all data contained in the tree.  This is intended for
 // tracking a delta enumeration reset, not for tree re-use, and will
-// cause the tree to flag itself as dirty in order to appropriately
+// cause the tree to flag itfolder as dirty in order to appropriately
```
Comment (ashmrtn) [inline src/internal/m365/collection/drive/delta_tree.go:70]: 🤔 not sure how this one popped out, but I think it's safe to revert this change 😉

File: src/internal/m365/collection/drive/delta_tree.go [hunk 2]
```diff
@@ -186,9 +191,8 @@ func (face *folderyMcFolderFace) setFolder(
 	if zombey, tombstoned := face.tombstones[id]; tombstoned {
 		delete(face.tombstones, id)
 
-		zombey.parent = parent
-		zombey.name = name
-		parent.children[id] = zombey
+		zombey.parent = parentNode
```
Comment (ashmrtn) [inline src/internal/m365/collection/drive/delta_tree.go:194]: should this also be updating the contained folder for the node like is done with the root above?

File: src/internal/m365/collection/drive/delta_tree_test.go [hunk 3]
```diff
@@ -65,98 +69,107 @@ func (suite *DeltaTreeUnitSuite) TestNewNodeyMcNodeFace() {
...
-			tname:     "add package",
-			tree:      treeWithRoot,
-			parentID:  rootID,
-			id:        folderID(),
-			name:      folderName(),
-			isPackage: true,
+			tname: "add package",
+			tree:  treeWithRoot,
+			folder: func() *custom.DriveItem {
+				return custom.ToCustomDriveItem(
```
Comment (ashmrtn) [inline src/internal/m365/collection/drive/delta_tree_test.go:108]: nit: since I think there's a helper can this be `custom.ToCustomDriveItem(d.pkgAtRoot())`?

File: src/internal/m365/collection/drive/delta_tree_test.go [hunk 4]
```diff
@@ -65,98 +69,107 @@ func (suite *DeltaTreeUnitSuite) TestNewNodeyMcNodeFace() {
...
+			},
 			expectErr: assert.Error,
 		},
 		{
-			tname:     "missing parentID",
-			tree:      treeWithRoot,
-			id:        folderID(),
-			name:      folderName(),
-			isPackage: true,
+			tname: "missing parentID",
```
Comment (ashmrtn) [inline src/internal/m365/collection/drive/delta_tree_test.go:141]: nit
```suggestion
			tname: "missing parent",
```

File: src/internal/m365/collection/drive/delta_tree_test.go [hunk 5]
```diff
@@ -168,30 +181,37 @@ func (suite *DeltaTreeUnitSuite) TestFolderyMcFolderFace_SetFolder() {
...
-			assert.Equal(t, test.isPackage, result.isPackage)
 
-			_, ded := tree.tombstones[test.id]
+			var (
+				expectID        = ptr.Val(folder.GetId())
+				expectName      = ptr.Val(folder.GetName())
+				expectIsPackage = folder.GetPackageEscaped() == nil
+				resultID        = ptr.Val(folder.GetId())
+				resultName      = ptr.Val(folder.GetName())
+				resultIsPackage = folder.GetPackageEscaped() == nil
```
Comment (ashmrtn) [inline src/internal/m365/collection/drive/delta_tree_test.go:202]: looks like copy/paste error here
```suggestion
				resultID        = ptr.Val(result.GetId())
				resultName      = ptr.Val(result.GetName())
				resultIsPackage = result.GetPackageEscaped() == nil
```

File: src/internal/m365/collection/drive/delta_tree_test.go [hunk 6]
```diff
@@ -453,11 +477,15 @@ func (ts tombs) compare(
...
+		expectID := ptr.Val(entombed.self.folder.GetId())
+
+		zombey := tombstones[expectID]
 		require.NotNil(t, zombey, "tombstone must exist")
 		assert.Nil(t, zombey.parent, "tombstoned node should not have a parent reference")
 
-		t.Run("assert_tombstones/"+zombey.id, func(_t *testing.T) {
+		resultID := ptr.Val(zombey.folder.GetId())
+
+		t.Run("assert_tombstones/"+resultID, func(_t *testing.T) {
```
Comment (ashmrtn) [inline src/internal/m365/collection/drive/delta_tree_test.go:488]: nit: I think something besides `/` in the test name would be better. It'll probably cause odd escaping errors when trying to run individual tests

**Review by ashmrtn — APPROVED** [review]

File: src/internal/m365/collection/drive/delta_tree.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/internal/m365/collection/drive/delta_tree.go:70]: no one expects the find&replace!

File: src/internal/m365/collection/drive/delta_tree.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/internal/m365/collection/drive/delta_tree.go:194]: good catch, yes.

File: src/internal/m365/collection/drive/delta_tree_test.go (hunk 3 above)
Comment (ryanfkeepers) [reply src/internal/m365/collection/drive/delta_tree_test.go:108]: right you are.

File: src/internal/m365/collection/drive/delta_tree_test.go (hunk 5 above)
Comment (ryanfkeepers) [reply src/internal/m365/collection/drive/delta_tree_test.go:202]: big oof
